web-common/src/features/sources/modal/submitAddDataForm.ts

```diff
@@ -142,3 +205,53 @@ export async function submitAddDataForm(
   // Go to the new connector file
   await goto(`/files/${newConnectorFilePath}`);
 }
+
+async function beforeSubmitForm(instanceId: string) {
+  // Emit telemetry
+  await behaviourEvent?.fireSourceTriggerEvent(
```

Comment:
Do we await everywhere we use telemetry? Perhaps it would be good to revisit this, if the end point is unreachable all the actions will be blocked.

Reply:
Good point. Removed.